src/main/java/com/zaxxer/hikari/pool/HikariPool.java

```diff
@@ -442,7 +442,16 @@ private PoolEntry createPoolEntry()
 
          final long maxLifetime = config.getMaxLifetime();
          if (maxLifetime > 0) {
-            final long variance = maxLifetime > 60_000 ? ThreadLocalRandom.current().nextLong(10_000) : 0;
+        	 long inHours = TimeUnit.MILLISECONDS.toHours(maxLifetime);
+        	 
+        	 // Every hour will add a 10 second random factor for the maxlifetime
+        	 long randomIntervalInMillis = inHours * 10_000;
+        	 
+        	 // Support the up to 10 seconds spread out when the lifetime is less than 1 hour
+        	 if (inHours == 0) {
```

Comment:
...reducing 90 sec from 30 mins of connections life (which is default) is not good :)
it should be calculated and 'restricted' with number of maximum connections in pool... eg:
default max  connections are 10 and if user sets their age to 1 day, then
... 'separating' age of 10 connections by 71 mins apart is too much.